### PR TITLE
Write e2e tests - Closes #196

### LINKED ISSUE #196: Write end-to-end tests with cucumber and protractor
This should be an exhaustive list of Lisk Explorer End-to-end test scenarios. We might not need to implement all of them right away, as some might be difficult to assert on (Activity graph) or set up in an isolated environment (Network monitor), But I want to have the list as a reference for thinking about e2e test coverage.
- [x] Homepage
    - [x] Top bar
        - [x] Search
            - [x] Block
            - [x] Transaction
            - [x] Address
            - [x] Delegate
            - [x] Not found
        - [x] Currency switcher
        - [x] Tools Menu
    - [x] Latest Transactions
    - [x] Latest Blocks
    - [x] Footer
        - [x] Decimal places selector
        - [x] Links
- [x] DB entity pages
    - [x] Account
        - [x] Summary
        - [x] Votes
            - [x] Link to delegate
        - [x] Transactions
            - [x] Link to transaction
            - [x] Link to sender
            - [x] Link to recipient
            - [x] Sent
            - [x] Received
            - [x] Load more
            - [x] Load less
    - [x] Delegate
        - [x] Summary
            - [x] Link to account
        - [x] Votes
            - [x] Link to account
        - [x] Voters
            - [x] Link to account
    - [x] Transaction
        - [x] Summary
        - [x] Details - TODO: What is the difference from Summary?
    - [x] Block
        - [x] Summary
        - [x] Transactions
            - [x] Load more
            - [x] Load less
- [x] Tools menu
    - [x] Top Accounts
        - [x] Link to account
        - [x] Load more
        - [x] Load less
    - [x] Activity Graph
        - [ ] Graph
        - [ ] Reset Camera
        - [x] Statistics
    - [x] Delegate Monitor
        - [x] Stats
        - [x] Latest votes
            - [x] Link to account
            - [x] Link to transaction
        - [x] Newest delegates
            - [x] Link to delegate
            - [x] Link to transaction
        - [x] Active Delegates
            - [x] Sorting
            - [x] Link to delegate
            - [x] Status tooltip
        - [x] Standby Delegates
            - [x] Link to delegate
            - [x] Next page
            - [x] Previous page
    - [x] Market Watcher
        - [x] Switch Poloniex/Bittrex
        - [x] Stats
        - [x] Hide Stats
        - [ ] Stock Chart
            - [ ] Minutes
            - [ ] Hours
            - [ ] Days
            - [ ] From/to change
            - [ ] Zoom change
        - [ ] Depth Chart
        - [x] Order Book
    - [x] Network Monitor
        - [x] Stats
        - [ ] Map
            - [ ] Zoom & pan to find markers
        - [ ] Connected peers
            - [ ] Sorting
        - [ ] Disconnected peers
            - [ ] Sorting
- [ ] All of the above with a smaller (mobile) screen

### PR DESCRIPTION
**Pull request by slaweet**
Covers checked points in #196 

Does **not** cover:
- activity graph
- market watcher charts
- network monitor
- tests on mobile screen

Those will be addressed in https://github.com/LiskHQ/lisk-explorer/issues/204

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -100,11 +136,16 @@ node('lisk-explorer-01'){
       sh '''#!/bin/bash
           # Stop Lisk-Node
           bash ~/lisk-test/lisk.sh stop
-          
+
           # Stop Lisk-Explorer
           pkill -f $(pwd)/app.js || true
+
+          # Cleanup e2e processes
+          pkill -f selenium -9 || true
```
Comment (Isabello) [inline Jenkinsfile]: we should create a cleanup function instead for all of these cleanup things (my fault)

Comment (slaweet) [inline Jenkinsfile]: @Isabello can you create this cleanup function? I have no idea how to do a function in Jenkinsfile.

**Comment by Isabello** [comment]
/api/search?id=6524861224470851795
/home/lisk/jenkins/workspace/Explorer-Pipeline_PR-205-22KSGGZHK6DQNFWMSX4336N535JVLIBMOP4P7YVPMBGT2CXNIYHQ/lib/api/delegates.js:378
                return error({ success : false, error : (body.error ? body.error : err) });

TypeError: Cannot read property 'error' of undefined
    at Request._callback (/home/lisk/jenkins/workspace/Explorer-Pipeline_PR-205-22KSGGZHK6DQNFWMSX4336N535JVLIBMOP4P7YVPMBGT2CXNIYHQ/lib/api/delegates.js:378:62)
    at self.callback (/home/lisk/jenkins/workspace/Explorer-Pipeline_PR-205-22KSGGZHK6DQNFWMSX4336N535JVLIBMOP4P7YVPMBGT2CXNIYHQ/node_modules/request/request.js:188:22)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at Request.onRequestError (/home/lisk/jenkins/workspace/Explorer-Pipeline_PR-205-22KSGGZHK6DQNFWMSX4336N535JVLIBMOP4P7YVPMBGT2CXNIYHQ/node_modules/request/request.js:884:8)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:188:7)
    at Socket.socketErrorListener (_http_client.js:309:9)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)

This is your failure causing tests to fail.

**Comment by slaweet** [comment]
I suppose the tests are failing because I changed test config `config.lisk.port = 7000` to  `config.lisk.port = 4000` and I didn't manage to convince Jenkins to run Lisk core on port 7000, though I tried to: https://github.com/LiskHQ/lisk-explorer/pull/205/commits/53fac6ee6f5e7bdb0c786520ed2db08936521438

Any advice on how to set it up @Isabello ?

**Comment by reyraa** [comment]
nice work Vit. thank you

**Pull request merged**